src/client/transport/transport-in-socket.ts

```diff
@@ -18,17 +20,23 @@ export default class TransportInSocket extends TransportBase {
         // NOTE: There is no special logic here.
     }
 
-    public queuedAsyncServiceMsg (msg: ServiceMessage): Promise<any> {
+    public queuedAsyncServiceMsg (msg: WebSocketServiceMessage): Promise<any> {
         return this.asyncServiceMsg(msg);
     }
 
-    public asyncServiceMsg (msg: ServiceMessage): Promise<any> {
+    public asyncServiceMsg (msg: WebSocketServiceMessage): Promise<any> {
+        const id = this._idGenerator.increment();
+
         return new Promise((resolve, reject) => {
             const handleMessage = (event) => {
-                cleanListeners();
                 this._activeServiceMsgCount--;
 
-                resolve(parse(event.data));
+                const data = parse(event.data);
+
+                if (data.id === id) {
```

Comment:
Decrement `this.activeServiceMsgCount` also should be under condition. And in that way will be better to use opposite condition block and just use `return` in it.